File: 백준/Silver/1747. 소수＆팰린드롬/소수＆팰린드롬.java
import java.util.Scanner;

public class Main {

    public static boolean isPrime(int x) {
        if (x < 2) {
            return false;
        }

        for (int i = 2; i <= Math.sqrt(x); i++) {
            if (x % i == 0) {
                return false;
            }
        }

        return true;
    }

    public static boolean isPalindrome(int x) {
        String str = Integer.toString(x);
        int len = str.length();

        for (int  i = 0; i < len / 2; i++) {
            if (str.charAt(i) != str.charAt(len - i - 1)) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        while (true) {
            if (isPalindrome(n) && isPrime(n)) {
                System.out.println(n);
                break;
            }
            n++;
        }
        sc.close();
    }
}

File: 백준/Silver/1747. 소수＆팰린드롬/소수＆팰린드롬.py
import math


# 소수 판별
def is_prime(x):
    if x < 2:
        return False

    for i in range(2, int(math.sqrt(x)) + 1):
        if x % i == 0:
            return False
    return True


# 팰린드롬 판별
def is_palindrome(x):
    return str(x) == str(x)[::-1]


n = int(input())

while True:
    if is_palindrome(n) and is_prime(n):
        print(n)
        break
    n += 1